test/e2e/federated-ingress.go

```diff
@@ -484,3 +513,119 @@ func getFederatedIngressAddress(client *fedclientset.Clientset, ns, name string)
 	}
 	return addresses, nil
 }
+
+// getNodeNames returns the names of nodes in a cluster.
+func getZoneNodesMap(clientset *kubeclientset.Clientset) (map[string][]string, error) {
+	zoneNodes := make(map[string][]string)
+	nodes, err := clientset.Core().Nodes().List(v1.ListOptions{})
+	if err != nil {
+		return nil, err
+	}
+	// Only consider schedulable nodes, because we won't
+	// have backends in non-schedulable nodes and don't
+	// want to open port to those nodes.
+	// Also, master node is unschedulable and contains
```

Comment:
We should not assume here that master node is unschedulable. Using Spec.Unschedulable should be enough